eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/persistence/DbsnpMonotonicAccessionGenerator.java

```diff
@@ -0,0 +1,56 @@
+package uk.ac.ebi.eva.accession.core.persistence;
+
+import uk.ac.ebi.ampt2d.commons.accession.core.AccessionWrapper;
+import uk.ac.ebi.ampt2d.commons.accession.core.SaveResponse;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionIsNotPendingException;
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicRangePriorityQueue;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+
+import java.util.List;
+import java.util.Map;
+
+public class DbsnpMonotonicAccessionGenerator<T> extends MonotonicAccessionGenerator {
+
+    public DbsnpMonotonicAccessionGenerator(long blockSize, String categoryId, String applicationInstanceId,
+                                            ContiguousIdBlockService contiguousIdBlockService) {
+        super(blockSize, categoryId, applicationInstanceId, contiguousIdBlockService);
+    }
+
+    @Override
+    public synchronized void recoverState(long[] committedElements) throws AccessionIsNotPendingException {
+        throw new UnsupportedOperationException("New accessions should not be generated");
```

Comment:
This message is too generic. I would add a reference that dbSNP accessions are the ones that we cannot generate. 

"New accessions cannot be issued for dbSNP variants"